### PR TITLE
Implement PartShop.search

### LINKED ISSUE #25: 'PartShop' object has no attribute 'search'
The [SBOL tutorial](https://github.com/SynBioDex/Community-Media/blob/master/2019/IWBDA19/workshop/solution.ipynb) uses PartShop.search(), which is not defined yet.

This is required for #14

### PR DESCRIPTION
**Pull request by tcmitchell**
Enough of PartShop.search is now implemented to support the tutorial. Exact searching and
general searching are both supported. Advanced searching (#240) will come later.

Fixes #25

### CODE REVIEW

File: sbol2/partshop.py [hunk 1]
```diff
@@ -323,3 +326,75 @@ def attachFile(self, top_level_uri, filepath):
         msg = 'HTTP Error code {} trying to attach file.'
         msg = msg.format(response.status_code)
         raise SBOLError(msg, SBOLErrorCode.SBOL_ERROR_BAD_HTTP_REQUEST)
+
+    def _make_search_item(self, item:dict) -> Identified:
```
Comment (bbartley) [inline sbol2/partshop.py:330]: I think you might be able to leverage this for the advanced search -> the SearchQuery is essentially a dict-like object that allows the user to specify arbitrary search parameters (e.g., RDF predicates and objects)

File: sbol2/partshop.py [hunk 2]
```diff
@@ -323,3 +326,75 @@ def attachFile(self, top_level_uri, filepath):
...
+        obj = Identified()
+        obj.identity = item['uri']
+        obj.displayId = item['displayId']
+        obj.name = item['name']
+        obj.description = item['description']
+        obj.version = item['version']
+        return obj
+
+    def _search(self, url:str) -> List[Identified]:
```
Comment (bbartley) [inline sbol2/partshop.py:339]: Would it be helpful to implement a generic method for structuring get requests?

There is a method `http_get_request` in libSBOL that does this:
https://github.com/SynBioDex/libSBOL/blob/master/source/partshop.cpp#L1060

It was added to the library relatively late, so many PartShop methods don't actually take advantage of it.  An advantage of a generic method is that it enables a single code change to update of all the methods if, for example, Synbiohub changes its http headers.

Comment (tcmitchell) [reply sbol2/partshop.py:339]: Yes, absolutely. I've been contemplating that. I think we want that as we [implement more part shop endpoints](#180)

File: sbol2/partshop.py (hunk 1 above)
Comment (tcmitchell) [reply sbol2/partshop.py:330]: This method gets run on the response to build the Identified instances for return to the client. I think what you're talking about needs to go at the beginning when converting the SearchQuery to a URL to query SynBioHub

**Pull request merged**